device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1418,7 +1419,7 @@ private void handleMessageException(IotHubTransportPacket packet, TransportExcep
                 RetryDecision retryDecision = config.getRetryPolicy().getRetryDecision(packet.getCurrentRetryAttempt(), transportException);
                 if (retryDecision.shouldRetry())
                 {
-                    this.taskScheduler.schedule(new MessageRetryRunnable(this.waitingPacketsQueue, packet, this), retryDecision.getDuration(), MILLISECONDS);
+                    this.taskScheduler.schedule(new MessageRetryRunnable(this.waitingPacketsQueue, packet, this.sendThreadLock), retryDecision.getDuration(), MILLISECONDS);
```

Comment:
This fixes a bug where requeued messages didn't wake up the send thread. They'd have to wait for the next queued message to come along from the user first. We were always supposed to pass in the sendThreadLock here, but somehow ended up just passing in ```this```